tests/backup/backup_helper.go

```diff
@@ -1348,3 +1350,119 @@ func restoreSuccessCheck(restoreName string, orgID string, retryDuration int, re
 	log.Infof("[%s] restored successfully", restoreName)
 	return restoreStatus, nil
 }
+
+// CreateCustomRestoreWithPVCs function can be used to deploy custom deployment with it's PVCs. It cannot be used for any other resource type.
+func CreateCustomRestoreWithPVCs(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string,
+	orgID string, ctx context.Context, storageClassMapping map[string]string, namespace string) (deploymentName string, err error) {
+
+	var bkp *api.BackupObject
+	var bkpUid string
+	var newResources []*api.ResourceInfo
+	var options metav1.ListOptions
+	var deploymentPvcMap = make(map[string][]string)
+	var deploymentNames []string
+	backupDriver := Inst().Backup
+	log.Infof("Getting the UID of the backup needed to be restored")
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+	if err != nil {
+		return "", err
+	}
+	for _, bkp = range curBackups.GetBackups() {
+		if bkp.Name == backupName {
+			bkpUid = bkp.Uid
+			break
+		}
+	}
```

Comment:
We can use `GetBackupUID` method from `drivers/backup/portworx/portworx.go`
This is repeated code

Reply:
Done